pyspecdata/core.py

```diff
@@ -4822,7 +4832,8 @@ def mod_BRD(guess,maxiter=20):
             logger.debug(strm('K dimlabels',K.dimlabels,'and raw shape',K.data.shape))
             self.reorder(dimname, first=False) # make the dimension we will be regularizing innermost
             logger.debug(strm('shape of the data is',ndshape(self),'is it fortran ordered?'))
-            data_fornnls = self.data.real
+            data_fornnls = self.data
+            demand_real(data_fornnls)
```

Comment:
This is more of a strategy/philosophy thing here --> I don't want to potentially "silently" convert the data to real -- instead, I try to raise a really explicit error message letting the user know what's going on, so they can do it themselves.